Update Reproducibility activities

cansavvy opened the pull request:
## Summary 

The Intermediate/Advanced reproducibility activity needs some updating

Trying to make sure headers and etc work in parallel with this https://docs.google.com/presentation/d/1_5FJ6SYA2t1F7BX5-Vvzi2LHEVUmXS4eFvousSqcl9w/edit#slide=id.g34609d90cb9_0_2226

kweav commented on reproducibility_adv_activity.Rmd:
```diff
@@ -28,19 +50,19 @@ After you install Docker, start up Docker desktop by double clicking on the app.
 
 ### Step 3: Run the docker image to start a container!
 
 <input type="checkbox"> Now if want to launch the image for use, we can use `docker run` to turn it into a **container**
```
```suggestion
<input type="checkbox"> Now if we want to launch the image for use, we can use `docker run` to turn it into a **container**
```

kweav commented on reproducibility_adv_activity.Rmd:
````diff
@@ -28,19 +50,19 @@ After you install Docker, start up Docker desktop by double clicking on the app.
+docker run jhudsl/reproducible-r
 ```
 
 <input type="checkbox"> To see what containers we have running we have we can run:
````
```suggestion
<input type="checkbox"> To see what containers we have running we can run:
```

kweav commented on reproducibility_adv_activity.Rmd:
```diff
@@ -73,35 +95,62 @@ This won't work because the file `run_analysis.sh` is not a file that our contai
 If we want to run `run_analysis.sh` there's two ways we could get our files on there.
 
 - We could download the files we need from online! We could `git clone` files from a repository or otherwise `wget` or `curl` files.
 - We could add a `volume` of files that are local to our computer. For this we will use the You'll `-v` or volume option.
```
```suggestion
- We could add a `volume` of files that are local to our computer. For this we will use the `-v` or volume option.
```

kweav commented on reproducibility_adv_activity.Rmd:
```diff
@@ -73,35 +95,62 @@ This won't work because the file `run_analysis.sh` is not a file that our contai
 
 If we want to run `run_analysis.sh` there's two ways we could get our files on there.
 
 - We could download the files we need from online! We could `git clone` files from a repository or otherwise `wget` or `curl` files.
```
```suggestion
- We could download the files we need from online! Options for this include using `git clone` for files from a repository or otherwise using `wget` or `curl` for files.
```

carriewright11 replied:
I cut this section to not give too much info on the checklist

kweav commented on reproducibility_adv_activity.Rmd:
```diff
@@ -73,35 +95,62 @@ This won't work because the file `run_analysis.sh` is not a file that our contai
 
-### Step 1: Let’s add our containers-for-scientists-sandbox files
-
 Point it so the volume has the workshop files
```
I'm not sure I understand this sentence.

carriewright11 replied:
I think this is no longer the title

kweav commented on reproducibility_adv_activity.Rmd:
```diff
@@ -73,35 +95,62 @@ This won't work because the file `run_analysis.sh` is not a file that our contai
-### Step 2: Let’s add our containers-for-scientists-sandbox files
+### Step 7: Let’s add our reproducibility-sandbox files
 
 We can specify a particular file path on our computer or give it `$PWD` Then optionally we can give a `:` and a file path we'd like this to be stored on on the container. Otherwise it will be stored at the absolute top of the container.
```
there are two on's here. "stored at on"? Not sure how to rephrase it

carriewright11 replied:
fixed this

kweav commented on reproducibility_adv_activity.Rmd:
```diff
@@ -73,35 +95,69 @@ This won't work because the file `run_analysis.sh` is not a file that our contai
+## Step 3: Take a look at your GitHub Action run!
 
-**Question: Does our container have all of the same software that our computer has?**
+<input type="checkbox"> You may need to refresh this page you are on.   
```
I think step 2 will take them to a different page?

kweav commented:
Left some suggestions for phrasing and comments if I was confused. The slides align well with the steps -- do you think having the screenshots included with `ottrpal::include_slide` would be helpful visualizing what people are supposed to?

carriewright11 commented on reproducibility_intro_activity.Rmd:
would be good where we start the pull request info to say where we are doing this - should say go to github desktop look for button at top that says current branch

carriewright11 replied:
adding this...

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -23,17 +23,22 @@ knitr::opts_chunk$set(number_sections = FALSE)
 ### Getting familiar with the repository
 
-This repository is a "mock" analysis. It has an example folder and file organizational structure that looks like this:
+This repository is a "mock" analysis. It has an example folder and file organizational structure that looks something like this:
```
updated below a tad @cansavvy

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,6 +77,8 @@ We will practice filing a pull request by filling in the the README. READMEs are
 
 But to create a pull request, we first need to create a branch to work from that is separate from the `main` copy.
 
+In GitHub Desktop:
```
added this @cansavvy

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
+<input type="checkbox"> Right click the file names to open up both the `01-heatmap.Rmd` file and the `README.md` file with any text editor. (May need to select `Other` and `Enable all applications` if on a Mac.)
 <input type="checkbox"> Copy all the text from `01-heatmap.Rmd` (Cmd + A - Mac, Ctrl + A - Windows).     
-<input type="checkbox"> Go to phind https://www.phind.com/search  
+<input type="checkbox"> Go to phind https://www.phind.com/search in your browser
```
Or 
```suggestion
<input type="checkbox"> Go to phind https://www.phind.com/search in your browser. (Other AI chatbots can also work like https://poe.com/)
```

carriewright11 replied:
adding this because phind was down when I tried :P

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
 
-<input type="checkbox"> Click the `Show in Finder` or `Show in Explorer` button depending on your operating system.  
-<input type="checkbox"> With any text editor open up both the `01-heatmap.Rmd` file and the `README.md` file  
+<input type="checkbox"> In GitHub Desktop, Click the `Show in Finder` or `Show in Explorer` button depending on your operating system.  This should be about halfway down the screen.
```
adding more details

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
 ### Adding changes to a branch
 
-<input type="checkbox"> Go back to GitHub Desktop. You should see `1 changed file` in the upper right corner. It may be upper left for Windows.    
+<input type="checkbox"> Go back to GitHub Desktop. You should see `1 changed file` in the upper right corner. It may be upper left depending on your machine and version of GitHub Desktop.    
```
more details

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
+<input type="checkbox"> Click on the `Files Changed` tab (lower set of tabs).   
+<input type="checkbox"> Click one of the plus signs to the left of one of your changes to leave a comment. Type something and then click `Add single comment`. Nice, you added a comment!
+<input type="checkbox"> Click another plus sign to the left of a change.
+<input type="checkbox"> This time, click the little symbol that has a plus and minus in a page (next to the preview tab). This will start a suggestion, where you can make suggested changes to documents. 
```
making it so people do a comment and a suggestion

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
-<input type="checkbox"> Return to the `Conversation` tab and scroll to the bottom. You should see an additional commit now!
-
-### 2b: Reading GitHub Actions logs
+<input type="checkbox"> Return to the `Conversation` tab and scroll to the bottom. You should see an additional commit and your comment now!
```
more info

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
-For more learning about using [GitHub Actions for science see our course](https://hutchdatascience.org/GitHub_Automation_for_Scientists/).
-
-### 2c: Updating your local repository
+### 2b: Updating your local repository
```
2b and 2c were in the opposite order in the slides - I think the slide version makes more sense

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
 <input type="checkbox"> Return to GitHub Desktop.  
 <input type="checkbox"> Click the `Fetch origin` button.  
-<input type="checkbox"> You should be prompted to `Pull origin` click this button.  
+<input type="checkbox"> You should be prompted to `Pull origin` click this button.  <input type="checkbox"> Click the History tab to see your changes over time. Note the most recent are at the top.
```
adding this to match slides

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
-<input type="checkbox"> You should be prompted to `Pull origin` click this button.  
+<input type="checkbox"> You should be prompted to `Pull origin` click this button.  <input type="checkbox"> Click the History tab to see your changes over time. Note the most recent are at the top.
 <input type="checkbox"> Now your changes should be synced. Open up the file you've changed with any default text editor to look for the change you made to make sure.  
+### 2c: Reading GitHub Actions logs
```
again just moving this down

carriewright11 commented on reproducibility_intro_activity.Rmd:
```diff
@@ -72,78 +77,85 @@ We will practice filing a pull request by filling in the the README. READMEs are
 <input type="checkbox"> Return to GitHub Pull Request Online.  
-<input type="checkbox"> Click the `Merge` button. Follow up with the `Confirm merge` button.  
+<input type="checkbox"> Click the `Merge` button. 
+<input type="checkbox"> Follow up with the `Confirm merge` button.  
```
adding these extra steps to match slides

carriewright11 commented on reproducibility_adv_activity.Rmd:
```diff
@@ -12,14 +12,36 @@ editor_options:
+### Installing Docker
+
+<input type="checkbox"> Go here to [install Docker](https://www.docker.com/get-started), following the instructions for your particular operating system.
+<input type="checkbox"> If you don't have a Docker account create an account when prompted, or [go here](https://hub.docker.com/).
```
```suggestion
<input type="checkbox"> If you don't have a Docker account yet, create an account when prompted, or [go here](https://hub.docker.com/).
```

carriewright11 commented:
OK @cansavvy , looks good to me! I added information about what was GUI vs command line to make it match the slides more. I also added details for things that might help people.

Two thoughts though-

For sake of time, do we want to prioritize showing people how to run rstudio? I am a tad worry it is going to take too much time to try to do everything. But maybe we assume we will not do the last activity?

Should we show people how to stop things? (I added a tiny bit about this)

cansavvy commented:
> OK @cansavvy , looks good to me! I added information about what was GUI vs command line to make it match the slides more. I also added details for things that might help people.

Thanks for adding! 

> For sake of time, do we want to prioritize showing people how to run rstudio? I am a tad worry it is going to take too much time to try to do everything. But maybe we assume we will not do the last activity?
> 
> Should we show people how to stop things? (I added a tiny bit about this) - 

This can be as quick as just showing them the delete button in the GUI

I was debating about taking out the part where we don't attach the volume first to save time. So I can do that. 

The last activity should be quick because we more or less are just navigating around github. 

I'll go through and see if there's other things we can shave in the interest of time. Thanks for doing this review.

carriewright11 commented:
OK @cansavvy This is all the changes we talked about.

carriewright11 commented:
<img width="853" alt="Screenshot 2025-04-03 at 10 11 39 PM" src="https://github.com/user-attachments/assets/2e69fb96-91bc-48d8-9f29-95dd042e641f" />

kweav commented on reproducibility_adv_activity.Rmd:
```diff
@@ -12,96 +12,339 @@ editor_options:
-<input type="checkbox"> We need to get the Docker **image** we want to use. We do this by "pulling" it.
 
-Run this command in your Terminal or Command Prompt window:
+<input type="checkbox">Search for the following image in the search box of Docker Desktop: `hudsl/reproducible-r`.
```
```suggestion
<input type="checkbox">Search for the following image in the search box of Docker Desktop: `jhudsl/reproducible-r`.
```
